homeassistant/components/alarmdecoder/alarm_control_panel.py

```diff
@@ -63,6 +64,8 @@ def _message_callback(self, message):
             self._state = STATE_ALARM_TRIGGERED
         elif message.armed_away:
             self._state = STATE_ALARM_ARMED_AWAY
+        elif message.armed_home and "***NIGHT-STAY***" in message.text:
```

Comment:
The alarmdecoder library should preferably expose the night stay mode as a separate arming mode attribute on the `message`.

We don't want to include device protocol in home assistant. That should be kept in the library.